hydro_lang/src/stream.rs

```diff
@@ -559,7 +578,13 @@ where
             HydroNode::CrossSingleton {
                 left: Box::new(self.ir_node.into_inner()),
                 right: Box::new(other.ir_node.into_inner()),
-                metadata: self.location.new_node_metadata::<(T, O2)>(),
+                metadata: self.location.new_node_metadata_with_kind::<(T, O2)>(
```

Comment:
Would be cool if `Stream::new` can validate the passed ordering... or better yet we could initialize with `None` and `Stream::new` could populate it. Don't love having two sources of truth.

Reply:
done!